src/hb-ot-shape.cc

```diff
@@ -851,7 +851,10 @@ hb_ot_substitute_default (const hb_ot_shape_context_t *c)
 
   HB_BUFFER_ALLOCATE_VAR (buffer, glyph_index);
 
-  _hb_ot_shape_normalize (c->plan, buffer, c->font);
+  if (buffer->message(c->font, "start normalize")) {
+    _hb_ot_shape_normalize (c->plan, buffer, c->font);
```

Comment:
Normalize cannot be skipped. It's where cmap mapping happens.

Reply:
OK; but skipping reordering at this point is useful. I'll move the tracing further down.

Reply:
Okay, yes, doing that in second (and third) stage of `*-normalize.cc` would be fine.